chart/templates/controller.yaml

```diff
@@ -52,52 +52,59 @@ rules:
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRoleBinding
 metadata:
-  name: {{.Values.wds_cp_name}}-transport-controller-rolebinding
+  name: {{.Values.wds_cp_name}}-transport-controller
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: ClusterRole
-  name: {{.Values.wds_cp_name}}-transport-controller-role
+  name: {{.Values.wds_cp_name}}-transport-controller
 subjects:
   - kind: ServiceAccount
-    name: transport-controller-sa
-    namespace: '{{.Values.wds_cp_name}}-system'
+    name: transport-controller
+    namespace: {{.Release.Namespace}}
 ---
 apiVersion: v1
 kind: ConfigMap
 metadata:
   name: transport-controller-config
-  namespace: {{.Values.wds_cp_name}}-system
+  namespace: {{.Release.Namespace}}
 data:
-  get-kubeconfig.sh: |
-    #!/bin/bash
-    # this script receives a ControlPlane name and a parameter
-    # that determines whether to extract the ControlPlane's in-cluster kubeconfig
-    # or the external kubeconfig (if set to "true", the first will be retrieved).
-    # The function returns the requested kubeconfig's content in base64.
-    # it assumes the kubeconfig context is set to access the hosting cluster.
-
-    cpname="$1"
-    get_incluster_key="$2"
-
-    key=""
-    if [[ "$get_incluster_key" == "true" ]]; then
-      key=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.inClusterKey}');
-    else
-      key=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.key}');
+  get-config.sh: |
+    #!/bin/env bash
+    cp_name="${1%"-system"}" # cp name or cp namespace
+    guess_its_name="$2"
+    if [ "$cp_name" == "" ] ; then
+      if [ "$guess_its_name" == "true" ] ; then
+        for cp in `kubectl get controlplane -o name`; do
+            cp=${cp##*/}

```

Comment:
@MikeSpreitzer can you please be more clear on what "this"  you are rereferring to?

Reply:
@MikeSpreitzer in my testing, it was returning the full name of the cp with the prefix.
I'm not sure why it would be different.
I don't think this is hurting so I would suggest to keep it to be conservative and cover my exception.

Reply:
Sorry, I was wrong about what `-o name` does. I find that it prefixes the name with the resource.apiGroup and a slash, which i think is what you are saying you saw too.

```shell
mspreitz@mjs13 kubestellar % kubectl --context kind-kubeflex get controlplane -o name
controlplane.tenancy.kflex.kubestellar.org/imbs1
controlplane.tenancy.kflex.kubestellar.org/wds1

mspreitz@mjs13 kubestellar % kubectl --context kind-kubeflex version       
Client Version: v1.28.7
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.29.2
```

So it looks like this prefix stripping is correct and necessary.